lib/bundler/ui/silent.rb

```diff
@@ -13,6 +17,7 @@ def confirm(message, newline = nil)
       end
 
       def warn(message, newline = nil)
+        @warnings |= [message]
```

Comment:
How do you feel about using a `Set` instead of the (esoteric and relatively confusing) `|=` operator?